src/AsmResolver.Workspaces/WorkspaceIndexNode.cs

```diff
@@ -81,5 +94,93 @@ public IEnumerable<T> GetRelatedObjects<T>(ObjectRelation<T> relation)
                 .Select(n => (T) n.Subject)
                 .Distinct();
         }
+
+        /// <summary>
+        /// Gets a stored data of type <see cref="T"/>.
+        /// </summary>
+        /// <param name="relation">The relation.</param>
+        /// <returns>default value if data with type <see cref="T"/> is not stored, otherwise the data.</returns>
+        /// <typeparam name="T">The type of data to obtain.</typeparam>
+        public T? GetData<T>() => Data
+            .OfType<T>()
+            .FirstOrDefault();
+
+        /// <summary>
+        /// Gets a stored data of type <see cref="T"/>.
+        /// </summary>
+        /// <param name="data">The data</param>
+        /// <returns>false if data with type <see cref="T"/> is not stored, otherwise true.</returns>
+        /// <typeparam name="T">The type of data to obtain.</typeparam>
+        public bool TryGetData<T>(out T? data)
```

Comment:
`data` parameter should probably have `[NotNullWhen(true)]`. While this is not available in .NET standard 2.0, we could consider adding it ourselves, similar to how it is done [here](https://www.thinktecture.com/en/net-core/c-8-nullable-reference-types-in-net-standard-21-and-net-4x/) or [here](https://www.meziantou.net/how-to-use-nullable-reference-types-in-dotnet-standard-2-0-and-dotnet-.htm).